### PR TITLE
Linux icon

### PR DESCRIPTION
**Pull request by fundies**
fix for #1066 and #694. Unfortunately, using custom icons would require a full path or installing your icon to /usr/share/icons (Something about it being insecure to open random images from relative paths). Neither of which is portable. Thus,  I've opted to hard code in an game icon I found in all my icon sets to the launcher. The user can edit the launcher later with any text editor if they prefer.

### CODE REVIEW

File: CompilerSource/compiler/compile.cpp [hunk 1]
```diff
@@ -712,6 +712,28 @@ wto << "namespace enigma_user {\nstring shader_get_name(int i) {\n switch (i) {\
   idpr("Closing game module and running if requested.",99);
   edbg << "Closing game module and running if requested." << flushl;
   fclose(gameModule);
+  
+  if (extensions::targetOS.identifier == "Linux" && mode == emode_compile)
+  {
```
Comment (JoshDreamland) [inline CompilerSource/compiler/compile.cpp]: Please extract this body into a separate method (eg, `write_desktop_entry()`).

File: install.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
```
Comment (JoshDreamland) [inline install.sh]: I'm assuming this is the change from your other PR. It'll go away if we check the other one in, first.

**Comment by time-killer-games** [comment]
Why not use the realpath() function for converting the relative path to an absolute?

**Comment by RobertBColton** [comment]
@time-killer-games Runtime is not the problem, to get a file to have an icon on Linux you create a desktop entry. It's basically just an ini file, and in that file you have a key for the `Icon=...` which has to be an absolute, not relative path. If we output a desktop entry, the user can't just put it into a ZIP and send it to people because then the absolute path to the icon is wrong if they put it in a different folder. This is basically the same problem GM has, and he's actually solving it the exact same way YoYo does I believe.

**Pull request closed without merging**